connectors/byoei.py

```diff
@@ -409,17 +444,61 @@ async def get_existing_ids(self, index):
             ts = doc["_source"].get(TIMESTAMP_FIELD)
             yield doc_id, ts
 
+    def done(self):
+        if self._fetcher_task is not None and not self._fetcher_task.done():
+            return False
+        if self._bulker_task is not None and not self._bulker_task.done():
+            return False
+        return True
+
+    async def cancel(self):
+        if self._fetcher_task is not None:
+            self._fetcher_task.cancel()
+            await self._fetcher_task
```

Comment:
need to try catch `CancelledError`

Reply:
Could you elaborate why?

The `asyncio.CancelledError` is caught in `SyncJobRunner.execute`, where it will try to call `elastic_server.cancel()`.

Reply:
if you don't catch it here you might not cancel the `bulker` task because it with raise and get off before.

Reply:
Do you mean
```python
if self._fetcher_task is not None:
    try:
        self._fetcher_task.cancel()
        await self._fetcher_task
    catch asyncio.CancelledError:
        pass
```